File: src/test/java/test/WebDriverFactory.java
package test;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.ie.InternetExplorerDriver;

public class WebDriverFactory {

    public static WebDriver createDriver(WebBrowserCode browser) {
        switch (browser) {
            case CHROME:
                return new ChromeDriver();
            case FIREFOX:
                return new FirefoxDriver();
            case EDGE:
                return new EdgeDriver();
            case IE:
                return new InternetExplorerDriver();
            default:
                throw new IllegalArgumentException("Unexpected driver code.");
        }
    }

}

File: src/test/java/test/WebDriverFactory.kt
package test

import org.openqa.selenium.WebDriver
import org.openqa.selenium.chrome.ChromeDriver
import org.openqa.selenium.edge.EdgeDriver
import org.openqa.selenium.firefox.FirefoxDriver
import org.openqa.selenium.ie.InternetExplorerDriver
import test.WebBrowserCode.*

fun createWebDriver(browserCode: WebBrowserCode): WebDriver =
    when (browserCode) {
        CHROME -> ChromeDriver()
        FIREFOX -> FirefoxDriver()
        EDGE -> EdgeDriver()
        IE -> InternetExplorerDriver()
        else -> throw IllegalArgumentException("Unexpected browser code.")
    }
